recipes/libdicom/all/conanfile.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+
+required_conan_version = ">=1.53.0"
+
+
+class LibdicomConan(ConanFile):
+    name = "libdicom"
+    description = "libdicom is a C library and a set of command-line tools for reading DICOM WSI files."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ImagingDataCommons/libdicom"
+    topics = ("dicom", "medical-imaging")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def requirements(self):
+        self.requires("uthash/2.3.0")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.2.3")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/2.0.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = MesonToolchain(self)
+        tc.project_options["tests"] = "false"
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate()
+
+    def build(self):
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
+        meson.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        fix_apple_shared_install_name(self)
+
+    def package_info(self):
+        self.cpp_info.libs = ["dicom"]
+        self.cpp_info.set_property("pkg_config_name", "libdicom")
+        if not self.options.shared:
+            self.cpp_info.defines = ["DCM_STATIC"]
```

Comment:
Sorry, I just double-checked it, and MesonToolchain is already defining the `default_library` in the built-in options section, so we can remove that part:

```suggestion
```

Reply:
Sorry, I'm confused. This exported define is not really related to MesonToolchain?

Reply:
Sorry, yes, it's related, and it's already defined by the `default_library` in the `conan_meson_xxxx.ini` within the `[built-in options]` section. See https://github.com/ImagingDataCommons/libdicom/blob/ba0e6d6f1fa08d69a4582b38817ca7ec699705e5/meson.build#L62C46-L62C61

Reply:
Discussed with @uilianries Keeping it to prevent missing symbols.